### PR TITLE
Cross-version API Compatibility - Cleanup Test Helper functions

### PR DESCRIPTION
**Pull request by mpanchajanya**
### What this PR does / why we need it

- Refactored the tests using updated test helpers.
- Added individual API test helpers for Servers and Context APIs to be reuse in test cases.

### Which issue(s) this PR fixes

Fixes #

### Describe testing done for PR
- Run Unit tests for test/compatibility
- Build and Run compatibility tests using below command and make sure all tests pass
`make compatibility-tests`
Screenshot 2023-03-22 at 23 31 22

More details on test output can be found from the Github workflow that is run on this PR https://github.com/vmware-tanzu/tanzu-plugin-runtime/actions/runs/4495795430/jobs/7909811037?pr=26

For more details on Cross-version API compatibility testing refer to https://github.com/vmware-tanzu/tanzu-plugin-runtime/blob/main/test/compatibility/docs/cross-version-api-compatibility.md

### Release note

```release-note

```

### CODE REVIEW

**Comment by vuil** [comment]
There are multiple references across PRs about the 3 make targets, apparently copied from
 test/compatibility/docs/cross-version-api-compatibility.md

If one needs to run the compatibility tests, do we need to run all 3? Seems the first covers the the other 2, no?
So I suggest updating  test/compatibility/docs/cross-version-api-compatibility.md to clearly state what one needs to run to execute the compatibility tests. You can still reference the other targets, but as sub headers or children of the main one.

Likewise in future PR. If `make compatibility-tests` is all that is needed to exercise the tests, just mention it, to reduce noise for the reviewers.

**Comment by vuil** [comment]
Some linter issues to address.

**Comment by mpanchajanya** [comment]
> There are multiple references across PRs about the 3 make targets, apparently copied from test/compatibility/docs/cross-version-api-compatibility.md
> 
> If one needs to run the compatibility tests, do we need to run all 3? Seems the first covers the the other 2, no? So I suggest updating test/compatibility/docs/cross-version-api-compatibility.md to clearly state what one needs to run to execute the compatibility tests. You can still reference the other targets, but as sub headers or children of the main one.
> 
> Likewise in future PR. If `make compatibility-tests` is all that is needed to exercise the tests, just mention it, to reduce noise for the reviewers.

You only need one Command make compatibility-tests. Updated the description and doc accordingly

**Comment by vuil** [comment]
This PR appears to be just refactoring, but I also noticed there are changes in the test result output? 
You might want to explain what is happening there as well. 
I also suggest updating the PR commits into logical commits with short details to explain the changes.
I would like to review how the commit message actually ends up at time of merge.

**Comment by mpanchajanya** [comment]
> Some linter issues to address.

Lints are addressed in https://github.com/vmware-tanzu/tanzu-plugin-runtime/pull/26

File: test/compatibility/core/api_constants.go [hunk 1]
```diff
@@ -63,7 +63,6 @@ const (
 	Version0254   RuntimeVersion = "v0.25.4"
 	Version0280   RuntimeVersion = "v0.28.0"
 	VersionLatest RuntimeVersion = "latest"
-	Version100    RuntimeVersion = "v1.0.0"
```
[the remark this answers is not in the document]
Comment (mpanchajanya) [reply test/compatibility/core/api_constants.go:66]: Removed all references for Version100. we will add this when we add tests for 1.0

**Review by vuil — APPROVED** [review]
lgtm, thanks for the updates

**Pull request merged**